scim2.yaml

```diff
@@ -1675,104 +1753,573 @@ paths:
+        <b>Scope(Permission) required:</b>
 
-#-----------------------------------------------------
-# Definitions
-#-----------------------------------------------------
-definitions:
-  #-----------------------------------------------------
-  # The User object
-  #-----------------------------------------------------
-  UserObject:
-    type: object
-    required:
-      - username
-      - password
-    properties:
-      schemas:
-        type: object
-        example: []
-      name:
-        type: object
-        properties:
-          firstName:
-            type: string
-            example: "Kim"
+            * internal_role_mgt_delete
+      operationId: deleteRoleV3
+      produces:
+        - application/scim+json
+      parameters:
+        - name: id
+          in: path
+          description: Unique ID of the resource type.
+          required: true
+          type: string
+      responses:
+        204:
+          description: Role is deleted
+        401:
+          description: Unauthorized
+          schema:
+            $ref: '#/definitions/ErrorUnauthorized'
+        403:
+          description: Forbidden
+          schema:
+            $ref: '#/definitions/ErrorForbidden'
+        404:
+          description: Valid role is not found
+          schema:
+            $ref: '#/definitions/ErrorNoRoleAvailable'
+        500:
+          description: Internal Server Error
+          schema:
+            $ref: '#/definitions/ErrorInternalServerError'
+    patch:
+      tags:
+        - Roles Endpoint
+      summary: Update Role - PATCH
+      description: |
+        This API updates the role's display name and permissions, and returns the updated role details using a PATCH operation.
+
+         <b>Scope(Permission) required:</b>
+
+             *
+      operationId: patchRole
+      consumes:
+        - application/scim+json
+      produces:
+        - application/scim+json
+      parameters:
+        - name: id
+          in: path
+          description: Unique id of the resource type.
+          required: true
+          type: string
+        - in: body
+          name: body
+          required: false
+          schema:
+            $ref: '#/definitions/PatchRoleOperationRequestObjectV3'
+      responses:
+        200:
+          description: Role is updated
+          schema:
+            $ref: '#/definitions/PatchRoleOperationResponseObjectV3'
+        401:
+          description: Unauthorized
+          schema:
+            $ref: '#/definitions/ErrorUnauthorized'
+        403:
+          description: Forbidden
+          schema:
+            $ref: '#/definitions/ErrorForbidden'
+        404:
+          description: Valid role is not found
+          schema:
+            $ref: '#/definitions/ErrorNoRoleAvailable'
+        406:
+          description: Not Acceptable
+          schema:
+            $ref: '#/definitions/ErrorNotAcceptable'
+
+  /v3/Roles/{id}/Users:
```

Comment:
As per the swagger PUT /v3/Roles/{id}/Users request looks like this:
Having /Users in the API path and having "users" key in the payload is redundant.
Can we change the payload to accept a JSON array? 
```
[
{
 "value" : "id1"
},
{
 "value" : "id2"
}
] 
```
<img width="894" alt="Screenshot 2025-06-20 at 20 12 48" src="https://github.com/user-attachments/assets/082acf8b-9f4d-4184-a2bd-d9f9d818606a" />

Reply:
Same concern applies for PATCH API call as well, check whether removing the users/groups key word from PATCH operations payload causes for complexity on payload parser

Reply:
Fixed with : https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/645/commits/273855a772841e31d731cff0fbe45cb738e7bbe4

Reply:
https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/645/commits/273855a772841e31d731cff0fbe45cb738e7bbe4 : The request payload should be sent without the attribute name. The API layer will add the attribute name and convert it to the required structure before sending it to the Charon layer.